### PR TITLE
uninstall swarm

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: docs/distributing-an-application/installing-with-swarm.md [hunk 1]
```diff
@@ -25,6 +25,90 @@ Quick install:
...
+```shell
+$ docker stack rm <stack_name>
+```
+
+for each stack with `replicated` in the name. Some portions of the stack deletion might fail, we will get back to those.
+
+__Services__
+
+```shell
+$ sudo docker service ls
```
Comment (stephenlawrence) [inline docs/distributing-an-application/installing-with-swarm.md]: Code snippets in bash should not include sudo.

Comment (ghost) [reply docs/distributing-an-application/installing-with-swarm.md]: Sure, removed

File: docs/distributing-an-application/installing-with-swarm.md [hunk 2]
```diff
@@ -25,6 +25,89 @@ Quick install:
...
+```
+
+for each stack with `replicated` in the name.
+
+
+__Lingering Containers__
+
+
+```shell
+$ docker ps
```
Comment (stephenlawrence) [inline docs/distributing-an-application/installing-with-swarm.md]: sb: docker ps -a

I had a stopped piper container that prevented me from removing a volume

**Comment by stephenlawrence** [comment]
Should we add the installed replicated files removal steps here as well (/var/lib/replicated, etc. like we have in our normal uninstall steps)?

File: docs/distributing-an-application/installing-with-swarm.md (hunk 2 above)
Comment (ghost) [reply docs/distributing-an-application/installing-with-swarm.md]: Good catch, addressed

**Comment by ghost** [comment]
Nope, those are all mounted as volumes

**Comment by ghost** [comment]
@marc @stephenlawrence any other feedback?

**Review by stephenlawrence — APPROVED** [review]

File: docs/distributing-an-application/installing-with-swarm.md [hunk 3]
```diff
@@ -66,6 +66,89 @@ export LC_CTYPE=C;echo "$(head -c 128 /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -
 docker stack deploy -c docker-compose.yml replicated
 ```
 
-## Installing Behind A Proxy
+### Installing Behind A Proxy
 
 Proxy support for Swarm will be included in a future release of Replicated.
+
+### Uninstall
```
Comment (emosbaugh) [inline docs/distributing-an-application/installing-with-swarm.md:73]: when i have some time we can maybe improve this with some bash foo

File: docs/distributing-an-application/installing-with-swarm.md [hunk 4]
```diff
@@ -66,6 +66,89 @@ export LC_CTYPE=C;echo "$(head -c 128 /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -
...
+__Services__
+
+```shell
+$ docker service ls
+```
+
+Using that output, run:
+
+```shell
+$ docker service rm <service_name>
```
Comment (emosbaugh) [inline docs/distributing-an-application/installing-with-swarm.md:99]:
```bash
sudo docker service ls | grep replicated | awk '{print $1}' | xargs docker service rm
```

File: docs/distributing-an-application/installing-with-swarm.md [hunk 5]
```diff
@@ -66,6 +66,89 @@ export LC_CTYPE=C;echo "$(head -c 128 /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -
...
+__Volumes__
+
+```shell
+$ docker volume ls
+```
+
+for each volume which is related to the Replicated onprem installation, run:
+
+```shell
+$ docker volume rm <volume_name>
```
Comment (emosbaugh) [inline docs/distributing-an-application/installing-with-swarm.md:153]:
```bash
sudo docker volume ls | grep replicated | awk '{print $2}' | xargs docker volume rm
```

File: docs/distributing-an-application/installing-with-swarm.md [hunk 6]
```diff
@@ -66,6 +66,89 @@ export LC_CTYPE=C;echo "$(head -c 128 /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -
...
+__Secrets__
+
+```shell
+$ docker secret ls
+```
+
+for each secret which is related to the Replicated onprem installation, run:
+
+```shell
+$ docker secret rm <secret_name>
```
Comment (emosbaugh) [inline docs/distributing-an-application/installing-with-swarm.md:141]:
```bash
sudo docker secret ls | grep replicated | awk '{print $1}' | xargs docker secret rm
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]
i dont think this should be merged the more i think about it.

**Comment by stephenlawrence** [comment]
@emosbaugh  Ok, do you mean not ever merged, or not until changes are applied?

**Comment by emosbaugh** [comment]
probably not in this state. if we cant get it better than this how about a script?

**Pull request closed without merging**